pages/shop/index.js

```diff
@@ -67,11 +69,30 @@ const Shop = (props) => {
   );
 };
 
+export async function getServerSideProps() {
+  //   context.res.setHeader('Cache-Control', `max-age=${CACHE_MAX_AGE}`);
+
+  const products = await getShopProducts();
+  // console.log(products);
+  return { props: { products } };
+}
+
 const mapStateToProps = (state) => {
   const shopListItemsCount = getShopListCount(state);
   return { shopListItemsCount };
 };
 
+Shop.propTypes = {
+  products: PropTypes.arrayOf(PropTypes.object),
+  addCartItem: PropTypes.func,
+  addShopListItem: PropTypes.func,
+  delCartItem: PropTypes.func,
+  delShopListItem: PropTypes.func,
```

Comment:
If all these props are not required, then we should have these included in the `defaultProps`

Reply:
But should we have placeholders of functions ?

Reply:
If these functions props are required in this component, make it as `.required`

Reply:
But a common track for component testing is to have two exports with and without redux's connect , just for testing props , and since functions are part of pre compiled code do we really need to make it is required?